resources/sip11/ra/src/main/java/org/mobicents/slee/resource/sip11/wrappers/DialogWrapper.java

```diff
@@ -501,9 +501,17 @@ public void delete() {
 		}
 		
 		final DialogState currentState = wrappedDialog != null ? wrappedDialog.getState() : null;
+		boolean isServerFlag = wrappedDialog != null ? wrappedDialog.isServer() : false; // is false ok, here?
 		boolean stackDoesNotFiresDialogTerminatedEvent = !isEnding()
-			&& !wrappedDialog.isServer() && (currentState == null || currentState == DialogState.TERMINATED);
-		wrappedDialog.delete();
+			&& !isServerFlag && (currentState == null || currentState == DialogState.TERMINATED);
```

Comment:
now seems right. As discussed in Gitter, please submit a new PR with unified changes in single commit. Thank you very much